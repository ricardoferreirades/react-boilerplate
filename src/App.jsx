import React from "react";
import "./App.css";

export const App = () => (
  <>
    <h1 className="title">Hello, boilerplate</h1>
  </>
);
