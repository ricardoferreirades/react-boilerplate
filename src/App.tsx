import React from "react";
import { Title } from "./App.styles";

function App() {
  const greeting: string = "Hello, Ricardo Gostoso, MARAVILHOSO!";
  return (
    <>
      <Title color="cyan">{greeting}</Title>
    </>
  );
}

export default App;
