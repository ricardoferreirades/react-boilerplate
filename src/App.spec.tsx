import React from "react";
import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import App from "./App";
import styled from "styled-components";
import renderer from "react-test-renderer";
import "jest-styled-components";

const Button = styled.button`
  color: green;
`;

describe("App Component", () => {
  let container: any = null;
  let root;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    act(() => {
      root.unmount(container);
    });
    container.remove();
    container = null;
  });

  it("works", () => {
    act(() => {
      root = createRoot(container);
      root.render(<App />);
    });
    expect(container.textContent).toBe("Hello, Ricardo Gostoso, MARAVILHOSO!");
  });

  it("works", () => {
    const tree = renderer.create(<Button>Foo</Button>);
    expect(tree).toMatchSnapshot();
  });
});
