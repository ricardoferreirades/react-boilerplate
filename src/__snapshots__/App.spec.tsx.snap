// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component works 1`] = `
.c0 {
  color: green;
}

<button
  className="c0"
>
  Foo
</button>
`;
